### ISSUE TITLE
Add new accounting method: Multiple depot method

### ISSUE THREAD

**Issue by Michael-A-Heuer** [rotki/rotki#2438]
## Abstract
Currently, all accounts are treated as a **single virtual depot** for the profit and loss calculation. 
The calculation should be possible for each depot individually as this can result in considerable tax savings (see screenshot below). This called the **multiple depot method** and is accepted, for example in Germany and the US.

## Motivation
Tax advisors in various jurisdictions argue that splitting coins into multiple separate accounts makes them more distinguishable/trackable. 
This is also found in German tax literature stating that the use of the FIFO method can be avoided by moving coins into a separate wallet: 
> „Falls die Anwendung des Fifo-Verfahrens nicht gewünscht ist, können u. E. etwaige
> Kryptowährungen „in eine Wallet verschoben werden“. Korrekt formuliert bedeutet dies, dass
> nicht die Kryptowährungen verschoben werden. Diese befinden sich weiterhin in der Blockchain.
> In der Wallet wird dann nur der Private Key gespeichert, der Zugang zu den mit diesem Key
> verbundenen Coins bietet. Ungeachtet dessen ist beim Halten von mehreren Coins eine genaue
> Zuordnung im Falle eines Verkaufs möglich. Die Fifo-Methode findet zudem keine Anwendung,
> wenn die Kryptowährungen (z. B. BTC) auf verschiedenen Exchanges gehalten werden. In diesen
> Fällen wird – wie bei einer Depottrennung bei Einkünften aus Kapitalvermögen – lediglich die
> Kryptowährung (die konkreten BTC) verkauft, die sich auf der Exchange befindet, auf der verkauft
> wird. Eine Vermischung findet daher u. E. nur auf der gleichen Exchange statt.“
(Heuel/Matthey, nwb 2018, 1037)

Accordingly, the tax calculation method (e.g. FIFO) can be applied to each account separately.
Other crypto tax services implement this feature as well (e.g. [accointing.com](https://www.accointing.com), [koinly.io](https://www.koinly.io)).

![image](https://user-images.githubusercontent.com/20623991/109124489-42afc480-774b-11eb-8855-ae2b210d1824.png)

**Comment by LefterisJP** [comment]
> This is also found in German tax literature stating that the use of the FIFO method can be avoided by moving coins into a separate wallet:

It's not that FiFo can be avoided. It's just that this applies FiFo separately to two or more depots.

I know of the depot method. I have heard people who used it in Germany and people who tried and couldn't since the Finanzamt complained. Seems to be random.

Thanks for making the issue. I know of at least one other user who asked for this in discord but did not make an issue in the end. I want us to implement this at some point. But as you can probably imagine it's not going to be trivial or easy. But it's in our radar.

**Comment by Michael-A-Heuer** [comment]
>It's not that FiFo can be avoided. It's just that this applies FiFo separately to two or more depots.

You are right of course. Literature sees it as 'avoided' in the sense that it is not applied across the depots.

**Comment by Michael-A-Heuer** [comment]
> > This is also found in German tax literature stating that the use of the FIFO method can be avoided by moving coins into a separate wallet:
> 
> It's not that FiFo can be avoided. It's just that this applies FiFo separately to two or more depots.
> 
> I know of the depot method. I have heard people who used it in Germany and people who tried and couldn't since the Finanzamt complained. Seems to be random.
> 
> Thanks for making the issue. I know of at least one other user who asked for this in discord but did not make an issue in the end. I want us to implement this at some point. But as you can probably imagine it's not going to be trivial or easy. But it's in our radar.

According to my tax advisor, the BMF has issued a **draft** for the tax offices, according to which the multi-deposit method will be officially allowed in Germany. The document has not been officially published yet, but one can expect it.

As shown in the picture, this can result in considerable tax savings and is supported by Accointing etc. already.

**Comment by LefterisJP** [comment]
I also spoke with another tax expert yesterday who also confirmed this is possible. Tax savings or not depends completely on the individual's case.

Sure this should be supported at some point. As has been stated before it's not trivial

**Comment by Michael-A-Heuer** [comment]
Some update on the current regulations in Germany on the use of the single-depot method currently used by Rotki to generate the P&L report.

I asked my German tax consultant who is specialized in cryptocurrencies if it is possible to switch from the multiple-depot method to the single-depot method

He pointed to the letter of the German federal ministry of finance from May 2022

> bb) Sequence of use
> [...]
> 61. The analysis is on a wallet-by-wallet basis. The selected method must be retained until all
units of the virtual currency or specific kind of other token in the wallet have been sold. The
method may be changed after the sale of all units of the virtual currency or specific kind of
other token in the wallet and the subsequent purchase of new units of the same virtual
currency or other token. If units of multiple virtual currencies or multiple kinds of other token
are held, the method may be elected separately for each virtual currency and each kind of
other token.
> (https://www.bundesfinanzministerium.de/Content/DE/Downloads/BMF_Schreiben/Steuerarten/Einkommensteuer/2022-05-09-einzelfragen-zur-ertragsteuerrechtlichen-behandlung-von-virtuellen-waehrungen-und-von-sonstigen-token-englische-version.pdf?__blob=publicationFile&v=3)

and said the following:

> In its BMF letter, the tax authorities prescribe a "wallet-based approach" (this probably means an address-based approach), i.e. the multiple depot method. According to the BMF letter from May 2022, a single-depot approach is not possible.

**Comment by LefterisJP** [comment]
As I responded to you in discord, this interpretation is mistaken. Naturally single depot is the standard way to do things as all your accounts are you. This is an approach verified by multiple accountans and experts we have worked with over the years. Please keep the issue for technical specification and or implementation details.

**Comment by LefterisJP** [comment]
Got a +1 on this from here: https://github.com/rotki/rotki/issues/6378

Request to maybe think of a better name for the method and multiple depot sounds too German-specific.

**Comment by Michael-A-Heuer** [comment]
> Got a +1 on this from here: #6378
> 
> Request to maybe think of a better name for the method and multiple depot sounds too German-specific.

That's right. I renamed this issue to "Add new accounting method: wallet-based cost tracking".

If there is any way to prioritize this request higher, I would be very grateful @LefterisJP. Despite being premium subscriber for years, I couldn't do a tax report with Rokti so far because I miss this feature. 
I have received confirmation from two independent German tax consultants (specialized for crypto) that due to the new German regulations, the wallet-based method must be used if you are using multiple wallets. Therefore, I think it is a must-have for German users.
I would rather like to spend the additional hundreds of Euros I am paying for other tax software products for Rotki.

@a7e8 mentioned:
>  It's probably also going to become the standard accounting method in many countries. Cointracker switched US customers to this method by default: https://support.cointracker.io/hc/en-us/articles/12407462337937 

so we should add the `🇺🇸 jurisdiction` label to this issue as well.

**Comment by LefterisJP** [comment]
> > Got a +1 on this from here: #6378
> > Request to maybe think of a better name for the method and multiple depot sounds too German-specific.
> 
> That's right. I renamed this issue to "Add new accounting method: wallet-based cost tracking".
> 
> If there is any way to prioritize this request higher, I would be very grateful @LefterisJP. Despite being premium subscriber for years, I couldn't do a tax report with Rokti so far because I miss this feature. I have received confirmation from two independent German tax consultants (specialized for crypto) that due to the new German regulations, the wallet-based method must be used if you are using multiple wallets. Therefore, I think it is a must-have for German users. I would rather like to spend the additional hundreds of Euros I am paying for other tax software products for Rotki.
> 
> @a7e8 mentioned:
> 
> > It's probably also going to become the standard accounting method in many countries. Cointracker switched US customers to this method by default: https://support.cointracker.io/hc/en-us/articles/12407462337937
> 

no way to prioritize this for our team at this point. This is a very difficult issue to implement due to its complicated nature. And as mentioned multiple times in discord and in social we are moving away from tax-reports as a main use-case of rotki. The PnL report and customizability will always be there, but it's too much work for all the jurisdicitons, too complicated and hard to monetize.

Instead we want rotki to become more of an active portfolio manager where you can actively manage your accounts and do trades from. And also expand to some other stuff out of crypto.

That said this does not mean we don't want to do it. The PnL report (note: **not tax report**) will be there,will be improved and this is a setting that should be added to it.

But we are also blocked by the complexity of the setting and the fact that we are switching to a unified history view while the PnL report is still using the old way of doing things. Once stuff get more streamlined, this feature will probably become easier to implement. But that's a lot of work preceding this.

> so we should add the `🇺🇸 jurisdiction` label to this issue as well.

Sure np. Will do.

**Comment by 7ql** [comment]
+1

I dearly need this to get off the dirty hands of ~~Accointing~~ Blockpit, Cointracker, Koinly and the like :(

**Comment by LefterisJP** [comment]
We are thinking of how this can be implemented. At least as far as I understand the idea here is that by moving the funds from one account to another the cost basis does not reset.

But it moves with the assets. Is this understanding correct?

-So say you got 2 BTC in Kraken and you bought them in January 2018.
- Then you move 1 to your ledger in February 2019
- Then you buy another BTC in Kraken in March 2020. So now you have 2 BTC in Kraken and 1 in your ledger.
- And then May 2020 you sell the 2 BTC in Kraken.

At that sell event the difference between the global method is that:

- With the global method the 2 BTC would be tax free since they were in a FiFo basis bought in 2018.
- With the wallet-based cost tracking the cost basis moves with the assets. So 1 of the Kraken BTC was bought in 2018. And the BTC in ledger was also bought in 2018. But you sell 2 BTC in Kraken in May 2020. 1 of them is the one from 2018, so tax free. The other you bought in March 2020, so less than year ago and then this one is taxable. But the 1 you have in ledger still keeps its tax free status.

Is that example what you would like to see achieved here?

**Comment by 7ql** [comment]
I'm very relieved and happy to read this. Yes, this is what I would like to see achieved.
Very much looking forward to it.

**Comment by heueristik** [comment]
Hey! I didn't check this issue for a long time.

> We are thinking of how this can be implemented. At least as far as I understand the idea here is that by moving the funds from one account to another the cost basis does not reset.
>
> But it moves with the assets. Is this understanding correct?

Yes, that's how blockpit (and others) are doing it as they write in their docs:

> The cost basis of an asset can also be transferred via the label [Transfer](https://intercom.help/blockpit/en/articles/12137159-transaction-label-transfer) to another integration.

(source: https://intercom.help/blockpit/en/articles/12137207-integration-based-cost-tracking-explained)

Note that they name wallets as "integrations".

**Comment by tkeitzl** [comment]
Hey guys, how is this coming along? Is https://github.com/rotki/rotki/issues/11097 really being shipped with [1.44.0](https://github.com/rotki/rotki/milestone/116) 🎉 🙏 ?

**Comment by LefterisJP** [comment]
Hey @tkeitzl can't promise ETAs. We are working on a bunch of issues that will allow unlocking this and others. It's part of a much bigger refactor we are doing. But yeah we want to achieve this as part of that change.  https://github.com/rotki/rotki/issues/6427

**Comment by aChrisYouKnow** [comment]
Having this done in advance of the October 15th tax deadline (for those who filled an extension for 2025 US taxes) would be huge.
